### DISCUSSION TITLE
Building TAO in CMake

### DISCUSSION THREAD

**Discussion by piperoc** [DOCGroup/ACE_TAO#2206] [Q&A]
I would love to test TAO, but I would prefer to use CMake because my sample solution uses CMake to target either x86 or ARM.

So if I can build form source in my CMake project, it would save me some headache going back and forth.

Unfortunately, I'm not the sharpest tool in the shed when it comes to complex builds like ACE/TAO; and I see there's a lot of configuration and code generation with the libraries.

I found that [the OpenDDS project](https://github.com/OpenDDS/OpenDDS) does just that: using CMake, it downloads and builds ACE?TAO from sources by default. 

I tried to reuse the generated libraries, but there's still a lot of hardcoded paths etc. that won't port easily to a different project. 

I was wondering if anyone has attempted to create a general CMake based build/install, or documented related guidelines.
Thanks

**Comment by mitza-oci** [comment]
> I would love to test TAO, but I would prefer to use CMake because my sample solution uses CMake to target either x86 or ARM.
> 
> So if I can build form source in my CMake project, it would save me some headache going back and forth.

You can build and install ACE+TAO using the documented processes.  They can then be used from your CMake project.


> Unfortunately, I'm not the sharpest tool in the shed when it comes to complex builds like ACE/TAO; and I see there's a lot of configuration and code generation with the libraries.
> 
> I found that [the OpenDDS project](https://github.com/OpenDDS/OpenDDS) does just that: using CMake, it downloads and builds ACE?TAO from sources by default.

OpenDDS is not building TAO using CMake.  Feel free to post a question there if you want to discuss further.

**Comment by piperoc** [comment]
Thank you for the quick reply, I will use the documented process and maintain separate builds for my targets.

I should have clearer on OpenDDS: what I noticed was that [OpenDDS has a CMake option](https://opendds.readthedocs.io/en/latest-release/devguide/building/index.html#cmake-building) to build eveything, including ACE/TAO (it looks like it runs scripts from CMake that call all kinds of configuration and build processes to make the libraries). All I had to do was clone the OpenDDS project, launch cmake on it and everything was done within 12 minutes. Including the libAce.a, libTao.a, etc.
I thought someone else might do something similar at a more generalized level, but I realize it is quite an undertaking...

**Comment by piperoc** [comment]
Actually, I dug deeper in the issues and found quite a number of similar requests. The best discussion is [here](https://github.com/DOCGroup/ACE_TAO/issues/266#issuecomment-1409620502) which I think supersedes my request. I will close this and defer the conclusion to those issues. Thanks again.
